hugr-py/tests/test_hugr_build.py

```diff
@@ -404,3 +405,47 @@ def test_option() -> None:
     dfg.set_outputs(b)
 
     validate(dfg.hugr)
+
+
+# a helper for the toposort tests
+def _build_test_hugrfn() -> Function:
+    f = Function("prepare_qubit", [tys.Bool, tys.Qubit])
+    [b, q] = f.inputs()
+
+    h = f.add_op(H, q)
+    q = h.out(0)
+
+    nnot = f.add_op(Not, b)
+
+    f.set_outputs(q, nnot, b)
+    return f
+
+
+# https://github.com/CQCL/hugr/issues/2350
+def test_toposort() -> None:
+    f = _build_test_hugrfn()
+
+    validate(Package([f.hugr], [QUANTUM_EXT]))
+
+    nodes = list(f.hugr)
+    func_node = nodes[1]
+
+    sorted_nodes = list(f.hugr.sort_region_nodes(func_node))
+    assert set(sorted_nodes) == {f.input_node, f.output_node, nodes[4], nodes[5]}
```

Comment:
I guess the "safer" way of checking this rather than indexing into the nodes array is something like
```suggestion
    assert set(sorted_nodes) == set(f.hugr.children(f))
```